src/transformers/models/depth_pro/modeling_depth_pro.py

```diff
@@ -0,0 +1,1083 @@
+
+        # STEP 3: apply patch and image encoder
+
+        patch_encodings = self.patch_encoder(
+            # each patch is processed as a separate batch
+            patches,
+            head_mask=head_mask,
+            # required for intermediate features
+            output_hidden_states=self.n_intermediate_hooks > 0,
+            return_dict=return_dict,
+        )
+
+        scaled_images_last_hidden_state = torch.split_with_sizes(patch_encodings[0], n_patches_per_scaled_image[::-1])
+        scaled_images_last_hidden_state = scaled_images_last_hidden_state[::-1]
+        # -1 (reverse list) as patch encoder returns high res patches first, we need low res first
+
+        # scale the image to patch size for image_encoder
+        image_scaled_to_patch_size = F.interpolate(
+            pixel_values,
+            size=(self.config.patch_size, self.config.patch_size),
+            mode="bilinear",
+            align_corners=False,
+        )
+        image_encodings = self.image_encoder(
+            pixel_values=image_scaled_to_patch_size,
+            head_mask=head_mask,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+        )
+
+        # calculate base height and width
+        # base height and width are the dimensions of the lowest resolution features
+        out_size = torch_int(image_encodings[0].shape[1] ** 0.5)
+        exponent_value = torch_int(math.log2(width / out_size))
+        base_height = height // 2**exponent_value
+        base_width = width // 2**exponent_value
+
+        # STEP 4: get patch features (high_res, med_res, low_res) - (3-5) in diagram
+
+        scaled_images_features = []
+        for i in range(self.n_scaled_images):
+            features = reconstruct_feature_maps(
+                scaled_images_last_hidden_state[i],
+                batch_size=batch_size,
+                padding=torch_int(self.merge_padding_value * (1 / self.scaled_images_ratios[i])),
+                output_size=(base_height * 2**i, base_width * 2**i),
+            )
+            scaled_images_features.append(features)
+
+        # STEP 5: get intermediate features - (1-2) in diagram
+
+        intermediate_features = []
+        for i in range(self.n_intermediate_hooks):
+            features = reconstruct_feature_maps(
+                # +1 to correct index position as hidden_states contain embedding output as well
+                patch_encodings[2][self.intermediate_hook_ids[i] + 1],
+                batch_size=batch_size,
+                padding=torch_int(self.merge_padding_value * (1 / self.scaled_images_ratios[-1])),
+                output_size=(
+                    base_height * 2 ** (self.n_scaled_images - 1),
+                    base_width * 2 ** (self.n_scaled_images - 1),
+                ),
+            )
+            intermediate_features.append(features)
```

Comment:
Please, do not use any ops in function/method call, create intermediate vars (maybe just super simple ones, but better to avoid it)

```suggestion
        for i in range(self.n_intermediate_hooks):
            # +1 to correct index position as hidden_states contain embedding output as well
            hidden_state = patch_encodings[2][self.intermediate_hook_ids[i] + 1]
            padding = torch_int(self.merge_padding_value * (1 / self.scaled_images_ratios[-1]))
            output_height = base_height * 2 ** (self.n_scaled_images - 1)
            output_width = base_width * 2 ** (self.n_scaled_images - 1)
            features = reconstruct_feature_maps(
                hidden_state
                batch_size=batch_size,
                padding=padding,
                output_size=(output_height, output_width)
            )
            intermediate_features.append(features)
```